src/worker/mod.rs

```diff
@@ -30,9 +36,17 @@ pub struct WorkerConfig {
     #[builder(default = "100")]
     pub max_outstanding_activities: usize,
     /// Maximum number of concurrent poll workflow task requests we will perform at a time on this
-    /// worker's task queue
+    /// worker's task queue. See also [WorkerConfig::nonsticky_to_sticky_poll_ratio]. Must be at
+    /// least 1.
     #[builder(default = "5")]
     pub max_concurrent_wft_polls: usize,
+    /// [WorkerConfig::max_concurrent_wft_polls] * this number = the number of max pollers that will
+    /// be allowed for the nonsticky queue when sticky tasks are enabled. If both defaults are used,
+    /// the sticky queue will allow 4 max pollers while the nonsticky queue will allow one. The
+    /// minimum for either poller is 1, so if `max_concurrent_wft_polls` is 1 and sticky queues are
+    /// enabled, there will be 2 concurrent polls.
+    #[builder(default = "0.2")]
```

Comment:
Is there any reason why we have 5:1 ratio here? Since by default all workflow tasks are sticky such default would lead to sub-optimal performance when sticky can not keep up and regular workers are idling. Should we remove this setting entirely and only control sticky cache size assuming 1:1 ratio between sticky and non sticky pollers?

Reply:
Talked with Max and we agreed it makes sense to prioritize sticky since that's where most tasks will be for a sticky worker. There is no situation where "regular workers" idle, because task queues sticky per-queue now, the same worker is always servicing both sticky and non sticky tasks. So it's purely a matter of which type of polling to prioritize.